### ISSUE TITLE
IllegalType Not Working For Annotation Using FQN

### ISSUE THREAD

**Issue by tntim96** [checkstyle/checkstyle#14798]
```bash
java -jar checkstyle-10.15.0-all.jar -c config.xml IllegalAnnotationClass.java
Starting audit...
Audit done.
```

```shell
cat config.xml
```

```xml
<!DOCTYPE module PUBLIC
    "-//Puppy Crawl//DTD Check Configuration 1.2//EN"
    "http://www.puppycrawl.com/dtds/configuration_1_2.dtd">

<module name="Checker">
  <module name="TreeWalker">
  <!--
    -->
    <module name="IllegalImport">
      <property name="illegalPkgs" value="testing"/>
      <property name="illegalClasses" value="testing.AnnotationClass"/>
    </module>
    <module name="IllegalType">
      <property name="illegalClassNames" value="testing.AnnotationClass"/>
    </module>
  </module>
</module>
```

```bash
cat IllegalAnnotationClass.java
```

```Java
public class IllegalAnnotationClass {

  @testing.AnnotationClass
  public void foo() {}
}
```
```bash
cat testing/AnnotationClass.java
```
```Java
package testing;

public @interface AnnotationClass {}
```

```bash
RUN_LOCALE="-Duser.language=en -Duser.country=US"
java $RUN_LOCALE -jar checkstyle-10.15.0-all.jar -c config.xml IllegalAnnotationClass.java
Starting audit...
Audit done.
```

---

Expected output
```
Starting audit...
[ERROR] /var/tmp/checkstyle-bug/IllegalAnnotationClass.java:3:4: Usage of type 'testing.AnnotationClass' is not allowed. [IllegalType]
Audit done.
Checkstyle ends with 1 errors.
````

**Comment by romani** [comment]
If you followed issue template , you might look at doc https://checkstyle.org/checks/coding/illegaltype.html#IllegalType

>For additional restriction of type usage see also: [IllegalInstantiation](https://checkstyle.org/checks/coding/illegalinstantiation.html#IllegalInstantiation), [IllegalImport](https://checkstyle.org/checks/imports/illegalimport.html#IllegalImport)

But I agree some clarity would be good to check description to define its targets.

Can other Checks help you with what you want to archive? or you need exactly this Check to handle it ?

**Comment by tntim96** [comment]
> Can other Checks help you

Not as far as I can tell. I think `IllegalType` is the most appropriate place for this check, however I noticed the documentation says:

> Specify classes that should not be used as types in variable declarations, return values or parameters.

...so strictly speaking, perhaps this is a feature request to expand the scope to include annotations.

**Comment by romani** [comment]
It might be related problem https://github.com/checkstyle/checkstyle/issues/6477
